D/D.sublime-syntax

```diff
@@ -8,339 +8,2032 @@ file_extensions:
+          push: [enum-member, type]
+    - include: not-whitespace-illegal
+
+  enum-member:
+    - match: '\b({{name}})\b'
+      scope: entity.name.constant.d
+      set:
+        - match: '='
+          scope: keyword.operator.assignment.d
+          set: [enum-member-after, value]
+        - match: ','
+          scope: punctuation.separator.sequence.d
+          pop: true
+        - match: '(?=})'
+          pop: true
+        - include: not-whitespace-illegal
+    - include: not-whitespace-illegal
+  enum-member-after:
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      pop: true
+    - match: '(?=})'
+      pop: true
+    - include: not-whitespace-illegal-pop
+
+  conditional-declaration-in:
+    - match: '\b(version)\b'
+      scope: keyword.control.conditional.d
+      push: [optional-block-statement-or-label, version-condition]
+    - match: '\b(debug)\b'
+      scope: keyword.control.conditional.d
+      push: [optional-block-statement-or-label, debug-condition]
+
+  version-condition:
+    - match: '\('
+      scope: punctuation.section.parens.begin.d
+      set:
+        - match: '\b(unittest|assert)\b'
+          scope: keyword.d
+          set: version-condition-end
+        - match: '\b({{name}})\b'
+          scope: constant.other.d
+          set: version-condition-end
+        - match: '{{integer_lookahead}}'
+          set: [version-condition-end, integer]
+        - include: not-whitespace-illegal
+    - include: not-whitespace-illegal
+  version-condition-end:
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      pop: true
+    - include: not-whitespace-illegal
+
+  debug-condition:
+    - match: '\('
+      scope: punctuation.section.parens.begin.d
+      set:
+        - match: '\b({{name}})\b'
+          scope: constant.other.d
+          set: version-condition-end
+        - match: '{{integer_lookahead}}'
+          set: [version-condition-end, integer]
+        - include: not-whitespace-illegal
+    - match: '(?=\S)'
+      pop: true
+
+  static-foreach-in:
+    - match: '\b(static)\s+(foreach|foreach_reverse)\b'
       captures:
-        1: punctuation.definition.comment.d
-  constant_placeholder:
-    - match: '(?i:%(\([a-z_]+\))?#?0?\-?[ ]?\+?([0-9]*|\*)(\.([0-9]*|\*))?[hL]?[a-z%])'
-      scope: constant.other.placeholder.d
-  function_assertions:
-    - match: '(in|out)\s*(?=\{)'
+        1: keyword.control.loop.d
+        2: keyword.control.loop.d
+      push: [optional-block-statement, foreach-arguments]
+
+  static-assert-in:
+    - match: '\b(static)\s+(assert)\b'
+      captures:
+        1: keyword.control.conditional.d
+        2: keyword.control.conditional.d
+      push: [expect-end-of-line, assert-arguments]
+
+  constructions-in:
+    - match: '{{construction_lookahead}}'
+      push: construction
+  construction:
+    # Post-blit constructor
+    - match: '\b(this)\s*(\()\s*(this)\s*(\))'
+      captures:
+        1: entity.name.function.post-blit.d
```

Comment:
Hmm, so is this defining a function named `this`? I'm not sure how practical/useful this will be in terms of the indexer, since I would imagine Goto Definition of `this` would list a lot. But perhaps this is related to that tmPreferences file for excluding constructors that I noted earlier in the review.

Reply:
Constructors are all defined through `this` and destructors through `~this`. Hence the exclusion in tmPreferences. It's still useful to have indexed within a file though, since a fair number of Devices files only contain a few constructors.